docs/en/operations/workload-scheduling.md

```diff
@@ -340,6 +340,38 @@ Blocked queries will wait indefinitely and not appear in `SHOW PROCESSLIST` unti
 
 Definitions of all workloads and resources in the form of `CREATE WORKLOAD` and `CREATE RESOURCE` queries are stored persistently either on disk at `workload_path` or in ZooKeeper at `workload_zookeeper_path`. ZooKeeper storage is recommended to achieve consistency between nodes. Alternatively `ON CLUSTER` clause could be used along with disk storage.
 
+## Configuration-based workloads and resources {#config_based_workloads}
+
+In addition to SQL-based definitions, workloads and resources can be predefined in the server configuration file. This is useful in cloud environments where some limitations are dictated by infrastructure, while other limits could be changed by customers. Configuration-based entities have priority over SQL-defined ones and cannot be modified or deleted using SQL commands.
+
+### Configuration format
+
+```xml
+<clickhouse>
+    <predefined_workloads>
```

Comment:
Instead of predefined_workloads and predefined_resources use combined resources_and_workloads and get rid of inner sql elements. Instead use plain text semicolon separated list of SQL statements, not split into individual XML elements, but as one string.

Make sure parsing code is updated as well. Reuse parsing code from workload entity keeper storage class.